framework/include/utils/PerfGraphRegistry.h

```diff
@@ -0,0 +1,90 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MooseTypes.h"
+
+// Forward Declarations
+class PerfGraph;
+class PerfGraphLivePrint;
+class PerfNode;
+
+namespace moose
+{
+namespace internal
+{
+
+// Forward Declarations
+class PerfGraphRegistry;
+
+PerfGraphRegistry & getPerfGraphRegistry();
```

Comment:
If this is gonna be global - should probably provide some info about it in a comment